modules/core/multihop/proof_gen.go

```diff
@@ -64,53 +62,34 @@ func (p ChanPath) GetConnectionHops() []string {
 	return hops
 }
 
-// GenerateProof generates a proof for the given key on the the source chain, which is to be verified on the dest
-// chain.
-func (p ChanPath) GenerateProof(
-	key []byte,
-	val []byte,
-	proofHeight exported.Height,
-	doVerify bool,
-) (result *channeltypes.MsgMultihopProofs, err error) {
-	defer func() {
-		if r := recover(); r != nil {
-			result = nil
-			err = sdkerrors.Wrapf(channeltypes.ErrMultihopProofGeneration, "%v", r)
-		}
-	}()
-
-	result = &channeltypes.MsgMultihopProofs{}
+// GenerateProof generates a proof for the given key on the the source chain, which is to be verified on the dest chain.
+func (p ChanPath) GenerateProof(key []byte, proofHeight exported.Height) (multihopProof channeltypes.MsgMultihopProofs, err error) {
 
 	// generate proof for key on source chain at the minimum consensus height known on the counterparty chain
-	maxProofHeight := p.source().Counterparty().QueryMaximumProofHeight(key, proofHeight, nil)
-	_, consensusHeightAB, err := p.source().Counterparty().QueryMinimumConsensusHeight(proofHeight, maxProofHeight)
+	maxProofHeight := p.source().QueryMaximumProofHeight(key, proofHeight, nil)
+	processedHeight, consensusHeight, err := p.source().Counterparty().QueryMinimumConsensusHeight(proofHeight, maxProofHeight)
```

Comment:
is processedHeight a chain height? If so maybe call it that to make it clearer (unless "processed" is standard)